### PR TITLE
List channels using conversations.list method

### PR DESCRIPTION
**Pull request by rnz0**
The conversations.list is method of the Conversations API that  overcomes the limitations of the legacy method `channels.list` to list channels as described in https://api.slack.com/methods/channels.list.
A cursor-based pagination method is introduced to retrieve large collections, in particular, list of public and private channels, multi-party IM and IM conversations.

For details see https://api.slack.com/methods/conversations.list

### CODE REVIEW

**Comment by mccartney** [comment]
@rnz0 can you please take care of the conflicts and then I can review and possibly merge?

File: src/main/scala/slack/api/SlackApiClient.scala [hunk 1]
```diff
@@ -271,6 +282,19 @@ class SlackApiClient private (token: String, slackApiBaseUri: Uri) {
     extract[Seq[Channel]](res, "channels")
   }
 
+  def listConversations(channelTypes: Seq[ConversationType] = Seq(PublicChannel), excludeArchived: Int = 0)(implicit system: ActorSystem): Future[Seq[Channel]] = {
```
Comment (mccartney) [inline src/main/scala/slack/api/SlackApiClient.scala]: Shouldn't `excludeArchived` be Boolean?

**Comment by rnz0** [comment]
@mccartney conflicts resolved, thanks.

**Comment by mccartney** [comment]
Thanks for the submission and for coming back with fixes/the new logic.

File: src/main/scala/slack/api/SlackApiClient.scala (hunk 1 above)
Comment (rnz0) [reply src/main/scala/slack/api/SlackApiClient.scala]: Fixed.

**Comment by rnz0** [comment]
For context, using this technique pagination of large collections eventually complete and takes from 10 to 15 minutes including multiple retries when Slack rate limits with 429 response code.

**Pull request merged**